diff --git a/pkg/pipelines/tekton/tasks.go b/pkg/pipelines/tekton/tasks.go
index e9cb0c47..787150ca 100644
--- a/pkg/pipelines/tekton/tasks.go
+++ b/pkg/pipelines/tekton/tasks.go
@@ -4,6 +4,8 @@ import (
 	"fmt"

 	pplnv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+	"knative.dev/func/pkg/openshift"
 )

 const (
@@ -13,10 +15,16 @@ const (
 )

 func taskFetchSources() pplnv1beta1.PipelineTask {
+	var taskKind = pplnv1beta1.NamespacedTaskKind
+	if openshift.IsOpenShift() {
+		taskKind = pplnv1beta1.ClusterTaskKind
+	}
+
 	return pplnv1beta1.PipelineTask{
 		Name: taskNameFetchSources,
 		TaskRef: &pplnv1beta1.TaskRef{
 			Name: "git-clone",
+			Kind: taskKind,
 		},
 		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{{
 			Name:      "output",
@@ -34,6 +42,7 @@ func taskBuildpacks(runAfter []string) pplnv1beta1.PipelineTask {
 		Name: taskNameBuild,
 		TaskRef: &pplnv1beta1.TaskRef{
 			Name: "func-buildpacks",
+			Kind: pplnv1beta1.ClusterTaskKind,
 		},
 		RunAfter: runAfter,
 		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{
@@ -78,6 +87,7 @@ func taskS2iBuild(runAfter []string) pplnv1beta1.PipelineTask {
 		Name: taskNameBuild,
 		TaskRef: &pplnv1beta1.TaskRef{
 			Name: "func-s2i",
+			Kind: pplnv1beta1.ClusterTaskKind,
 		},
 		RunAfter: runAfter,
 		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{
@@ -111,6 +121,7 @@ func taskDeploy(runAfter string, referenceImageFromPreviousTaskResults bool) ppl
 		Name: taskNameDeploy,
 		TaskRef: &pplnv1beta1.TaskRef{
 			Name: "func-deploy",
+			Kind: pplnv1beta1.ClusterTaskKind,
 		},
 		RunAfter: []string{runAfter},
 		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{{
