diff --git a/k8s/security_context.go b/k8s/security_context.go
index 7d2fd569..956c7623 100644
--- a/k8s/security_context.go
+++ b/k8s/security_context.go
@@ -9,14 +9,8 @@ import (
 var oneTwentyFour = semver.MustParse("1.24")
 
 func defaultPodSecurityContext() *corev1.PodSecurityContext {
-	// change ownership of the mounted volume to the first non-root user uid=1000
-	runAsUser := int64(1000)
-	runAsGroup := int64(1000)
-	return &corev1.PodSecurityContext{
-		RunAsUser:  &runAsUser,
-		RunAsGroup: &runAsGroup,
-		FSGroup:    &runAsGroup,
-	}
+	// for OpenShift [u|g|fs] ids are set automatically
+	return nil
 }
 
 func defaultSecurityContext(client *kubernetes.Clientset) *corev1.SecurityContext {
