diff --git a/k8s/dialer.go b/k8s/dialer.go
index d11116d3..0e53d052 100644
--- a/k8s/dialer.go
+++ b/k8s/dialer.go
@@ -148,7 +148,7 @@ func (c *contextDialer) startDialerPod(ctx context.Context) (err error) {
 					Stdin:           true,
 					StdinOnce:       true,
 					Command:         []string{"socat", "-u", "-", "OPEN:/dev/null"},
-					SecurityContext: defaultSecurityContext(),
+					SecurityContext: defaultSecurityContext(client),
 				},
 			},
 			DNSPolicy:     coreV1.DNSClusterFirst,
diff --git a/k8s/persistent_volumes.go b/k8s/persistent_volumes.go
index 0a73812d..da1caf1e 100644
--- a/k8s/persistent_volumes.go
+++ b/k8s/persistent_volumes.go
@@ -128,7 +128,7 @@ func runWithVolumeMounted(ctx context.Context, podImage string, podCommand []str
 							MountPath: volumeMntPoint,
 						},
 					},
-					SecurityContext: defaultSecurityContext(),
+					SecurityContext: defaultSecurityContext(client),
 				},
 			},
 			Volumes: []corev1.Volume{{
diff --git a/k8s/security_context.go b/k8s/security_context.go
index 2664efb5..346e7630 100644
--- a/k8s/security_context.go
+++ b/k8s/security_context.go
@@ -1,16 +1,34 @@
 package k8s
 
 import (
+	"strconv"
+
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-func defaultSecurityContext() *corev1.SecurityContext {
+func defaultSecurityContext(client *kubernetes.Clientset) *corev1.SecurityContext {
 	runAsNonRoot := true
-	return &corev1.SecurityContext{
+	sc := &corev1.SecurityContext{
 		Privileged:               new(bool),
 		AllowPrivilegeEscalation: new(bool),
 		RunAsNonRoot:             &runAsNonRoot,
 		Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
 		SeccompProfile:           nil,
 	}
+	if info, err := client.ServerVersion(); err == nil {
+		var maj, min int64
+		maj, err = strconv.ParseInt(info.Major, 10, 64)
+		if err != nil {
+			maj = -1
+		}
+		min, err = strconv.ParseInt(info.Minor, 10, 64)
+		if err != nil {
+			min = -1
+		}
+		if (maj >= 2) || (maj == 1 && min >= 24) {
+			sc.SeccompProfile = &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault}
+		}
+	}
+	return sc
 }
